jetstream/analysis.py

```diff
@@ -629,3 +633,47 @@ def ensure_enrollments(self, current_date: datetime) -> None:
             )
         except Conflict:
             pass
+
+    def export_errors(self, bucket_name: str):
+        """Export experiment errors to GCS."""
+        if self.config.experiment.normandy_slug is None:
+            return
+
+        if self.log_config is None:
+            log_project = self.project
+            log_dataset = self.dataset
+            log_table = "logs"
+        else:
+            log_project = self.log_config.log_project_id or self.project
+            log_dataset = self.log_config.log_dataset_id or self.dataset
+            log_table = self.log_config.log_table_id or "logs"
+
+        logger.info(f"Retrieving logs from BigQuery: {log_project}.{log_dataset}.{log_table}")
+
+        bq_log_client = BigQueryClient(log_project, log_dataset)
+        logs_df = bq_log_client.table_to_dataframe(log_table)
+
+        exp_logs = logs_df[logs_df["experiment"] == self.config.experiment.normandy_slug]
```

Comment:
It might be more performant to write a SQL query here with a filter on the experiment instead of loading the entire logs table for each analysis run. For now the table is relatively small, but it could potentially grow.

Another things we could do inside a query is to make sure that only the most recent errors for an experiment are selected. For example, if multiple reruns had been triggered for an experiments then older errors are no longer relevant and could be filtered out here.

We could event consider to cluster the logs table on  the `experiment` column.

Reply:
Yea I like this idea. I was originally trying to use existing functionality, but you're right that converting the entire table to a dataframe and manipulating from there is inefficient. I'll make some updates and send another review request when they're ready.